hera_cal/lstbin.py

```diff
@@ -256,9 +256,9 @@ def lst_bin(data_list, lst_list, flags_list=None, nsamples_list=None, dlst=None,
                     key = antpair + (pol,)
                     if key not in data and ((key[0] != key[1] and utils.reverse_bl(key) not in data) or key[0] == key[1]):
                         # last part lets us spoof ne and en for autocorrs. If we dont include it, only en xor ne will be spoofed.
-                        nsamples[key] = odict({ind: [] for ind in range(len(lst_grid))})
-                        data[key] = odict({ind: [] for ind in range(len(lst_grid))})
-                        flags[key] = odict({ind: [] for ind in range(len(lst_grid))})
+                        nsamples[key] = odict({ind: np.empty((0, Nfreqs), dtype=np.int8) for ind in range(len(lst_grid))})
+                        data[key] = odict({ind: np.empty((0, Nfreqs), dtype=np.complex64) for ind in range(len(lst_grid))})
```

Comment:
shouldn't this be complex128

Reply:
No. If you add a complex128 array to the vstack, it'll promote the type. But this way it doesn't promote until it's necessary, allowing you to use single precision if you want.

Reply:
done